src/scripts.c

```diff
@@ -51,7 +51,11 @@ static void update_triggers(void)
 {
 	__attribute__((unused)) int ret = 0;
 
-	ret = system("/usr/bin/systemctl --no-block daemon-reload");
+	if (need_systemd_reexec) {
+		ret = system("/usr/bin/systemctl daemon-reexec");
+	}
```

Comment:
However, doing a `reload` after a `reexec` is just duplicate, so the code should be:

```
if (need_systemd_reexec) {
        ret = system("/usr/bin/systemctl daemon-reexec");
} else {
        ret = system("/usr/bin/systemctl daemon-reload");
}
```